hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -1176,4 +1178,37 @@ public void testDatabaseLocationWithSlashInWarehouseDir() {
 
     Assert.assertEquals("s3://bucket/database.db", database.getLocationUri());
   }
+
+  @Test
+  public void testRegisterTable() {
+    TableIdentifier identifier = TableIdentifier.of(DB_NAME, "t1");
+    catalog.createTable(identifier, getTestSchema());
+    Table registeringTable = catalog.loadTable(identifier);
+    catalog.dropTable(identifier, false);
+    TableOperations ops = ((HasTableOperations) registeringTable).operations();
+    String metadataLocation = ((HiveTableOperations) ops).currentMetadataLocation();
+    Table registeredTable = catalog.registerTable(identifier, metadataLocation);
+    org.assertj.core.api.Assertions.assertThat(registeredTable).isNotNull();
+    TestHelpers.assertSerializedAndLoadedMetadata(registeringTable, registeredTable);
+    String expectedMetadataLocation =
+        ((HasTableOperations) registeredTable).operations().current().metadataFileLocation();
+    org.assertj.core.api.Assertions.assertThat(metadataLocation)
+        .isEqualTo(expectedMetadataLocation);
+    org.assertj.core.api.Assertions.assertThat(catalog.loadTable(identifier)).isNotNull();
+    org.assertj.core.api.Assertions.assertThat(catalog.dropTable(identifier)).isTrue();
+  }
+
+  @Test
+  public void testRegisterExistingTable() {
+    TableIdentifier identifier = TableIdentifier.of(DB_NAME, "t1");
+    catalog.createTable(identifier, getTestSchema());
+    Table registeringTable = catalog.loadTable(identifier);
+    TableOperations ops = ((HasTableOperations) registeringTable).operations();
+    String metadataLocation = ((HiveTableOperations) ops).currentMetadataLocation();
+    org.assertj.core.api.Assertions.assertThatThrownBy(
+            () -> catalog.registerTable(identifier, metadataLocation))
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessage("Table already exists: hivedb.t1");
+    org.assertj.core.api.Assertions.assertThat(catalog.dropTable(identifier, true)).isTrue();
```

Comment:
Instead of ugly static import, can we use the same Assertion here as the other tests (there doesnt seem anything specific about this that needs a different Assertion)?

Reply:
I guess alternatively, we could keep this ugly class reference, but we need to make sure to make a follow up PR to clean up the rest of TestHiveCatalog's assertions to use "org.assertj.core.api.Assertion"?

Reply:
`org.junit.jupiter.api.Assertions` from another package was imported already, which is why IDE automatically used full class reference `org.assertj.core.api.Assertions` while pasting the code.

I have now imported `org.assertj.core.api.Assertions.assertThat` static in the class.